components/org.wso2.analytics.is/src/main/test/org/wso2/analytics/is/siddhi/extension/JsonFlatMapEventDuplicatorStreamProcessorTestCase.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.analytics.is.siddhi.extension;
+
+import junit.framework.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.wso2.siddhi.core.ExecutionPlanRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.query.output.callback.QueryCallback;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.util.EventPrinter;
+
+public class JsonFlatMapEventDuplicatorStreamProcessorTestCase {
+    private int inEventCount;
+    private int removeEventCount;
+    private boolean eventArrived;
+
+    @Before
+    public void init() {
+        inEventCount = 0;
+        removeEventCount = 0;
+        eventArrived = false;
+    }
+
+    @Test
+    public void eventDuplicatorDefaultDelimiterTest1() throws InterruptedException {
```

Comment:
Shall we explain on what we are actually doing with this test? Why we have 1 at the end, are we having multiple test cases?